sycl/source/device_selector.cpp

```diff
@@ -30,28 +30,36 @@ static bool isDeviceOfPreferredSyclBe(const device &Device) {
 
 device device_selector::select_device() const {
   vector_class<device> devices = device::get_devices();
-  int score = -1;
+  int score = REJECT_DEVICE_SCORE;
   const device *res = nullptr;
+
   for (const auto &dev : devices) {
     int dev_score = (*this)(dev);
+
     if (detail::pi::trace(detail::pi::TraceLevel::PI_TRACE_ALL)) {
       string_class PlatformVersion = dev.get_info<info::device::platform>()
                                          .get_info<info::platform::version>();
       string_class DeviceName = dev.get_info<info::device::name>();
       std::cout << "SYCL_PI_TRACE[all]: "
-                << "select_device(): -> score = " << score << std::endl
+                << "select_device(): -> score = " << score
+                << ((score == REJECT_DEVICE_SCORE) ? "(REJECTED)" : " ")
+                << std::endl
                 << "SYCL_PI_TRACE[all]: "
                 << "  platform: " << PlatformVersion << std::endl
                 << "SYCL_PI_TRACE[all]: "
                 << "  device: " << DeviceName << std::endl;
     }
 
+    // Device is discarded if is marked with REJECT_DEVICE_SCORE
+    if (dev_score == REJECT_DEVICE_SCORE)
```

Comment:
I think it is not quite correct. The SYCL spec(4.6.1.1 Device selector interface) says:
`If a negative score is returned then the corresponding SYCL device will never be chosen.`
So if a user provides a custom selector which returns -2 this issue will be still here.
I've created a PR which resolves the same issue: https://github.com/intel/llvm/pull/1751.
Please, let me know if you want to fix the issue in your PR or we commit 1751.

Reply:
@romanovvlad, are you okay if we merge this PR and rebase #1751?

Reply:
I would prefer that we do not merge incorrect implementation.

Reply:
I don't think this patch makes it worse than it is today.

Reply:
I don't think either.